### ISSUE TITLE
Ensure jq is installed before rook-ceph mons restore-quorum is started

### ISSUE THREAD

**Issue by mitchese** [rook/kubectl-rook-ceph#81]
After freshly installing via Krew, some dependencies are not identified as missing or required in order to run commands. 

During a scary recovery of mon  quorum, I discovered that JQ was required (but this after it starts running the recovery)

The following is a log of what happened: 

```
:) $ kubectl rook-ceph mons restore-quorum d
mon=b, endpoint=...:6789
mon=d, endpoint=...:6789
mon=c, endpoint=...:6789
Info: Check for the running toolbox
Info: Waiting for the pod from deployment "rook-ceph-tools" to be running
deployment.apps/rook-ceph-tools condition met

Warning: Restoring mon quorum to mon d (...)
Info: The mons to discard are: b c
Info: The cluster fsid is 80....7
Info: Are you sure you want to restore the quorum to mon "d"? If so, enter: yes-really-restore
yes-really-restore
Info: proceeding
deployment.apps/rook-ceph-operator scaled
deployment.apps/rook-ceph-mon-b scaled
deployment.apps/rook-ceph-mon-c scaled
deployment.apps/rook-ceph-mon-d scaled
Info: Waiting for operator and mon pods to stop
pod/rook-ceph-mon-d-57554c46f8-9xcw2 condition met
setting debug mode for "rook-ceph-mon-d"
/******/.krew/bin/kubectl-rook_ceph: line 654: jq: command not found
```

This is a very bad/scary time to hit this issue. I was able to recover by scaling back mon-d to one replica, installing jq and re-starting the restore-quorum.

What should have happened:
Ideally during install, it pulls in dependencies with it. I don't think this is possible with krew installs, so at least before starting a restore-quorum it should check if all dependencies are available, and only once we are sure we can proceed, then it should start with recovery. 

```
:) $ kubectl rook-ceph mons restore-quorum d
mon=b, endpoint=...:6789
mon=d, endpoint=...:6789
mon=c, endpoint=...:6789
Info: Check for the running toolbox
Info: Waiting for the pod from deployment "rook-ceph-tools" to be running
deployment.apps/rook-ceph-tools condition met
** CHECKING DEPENDENCIES HERE: ERROR JQ NOT INSTALLED OR REQUIRED ** 

Please install jq first and re-run... No changes have been made.
```

or so.

**Comment by subhamkrai** [comment]
@mitchese could you install jq and try again? we didn't think of this will fix this.

**Comment by mitchese** [comment]
Yes that's exactly what I ended up doing. I updated the original issue to reflect this.

Simply installing jq does work and then it can be re-started, but oof that was quite the scare in an already scary situation

**Comment by subhamkrai** [comment]
> Yes that's exactly what I ended up doing. I updated the original issue to reflect this.
> 
> Simply installing jq does work and then it can be re-started, but oof that was quite the scare in an already scary situation

good to know it is working for you and everything healthy now, we'll fix this in next release

**Comment by subhamkrai** [comment]
This issue is not valid now, since we moved to go based script. Thanks!

**Comment by travisn** [comment]
Seems like we still need jq [here](https://github.com/rook/kubectl-rook-ceph/blob/master/pkg/rook/status.go#L28). It would be great to remove this dependency.

**Comment by subhamkrai** [comment]
> Seems like we still need jq [here](https://github.com/rook/kubectl-rook-ceph/blob/master/pkg/rook/status.go#L28). It would be great to remove this dependency.

right, I missed that. I was thinking how we can do that seems like it's not easy if we want to replicate the current output. The closest I can think will print 
```yaml
 get cephcluster -o jsonpath='{range .items[*]}{.status}{"\n"}{end}'
{"ceph":{"capacity":{"bytesAvailable":20949798912,"bytesTotal":20971520000,"bytesUsed":21721088,"lastUpdated":"2023-06-23T15:35:08Z"},"fsid":"cd10c270-c912-4e67-ab86-b4161df498d1","health":"HEALTH_OK","lastChecked":"2023-06-23T15:35:08Z","versions":{"mgr":{"ceph version 17.2.6 (d7ff0d10654d2280e08f1ab989c7cdf3064446a5) quincy (stable)":1},"mon":{"ceph version 17.2.6 (d7ff0d10654d2280e08f1ab989c7cdf3064446a5) quincy (stable)":1},"osd":{"ceph version 17.2.6 (d7ff0d10654d2280e08f1ab989c7cdf3064446a5) quincy (stable)":1},"overall":{"ceph version 17.2.6 (d7ff0d10654d2280e08f1ab989c7cdf3064446a5) quincy (stable)":3}}},"conditions":[{"lastHeartbeatTime":"2023-06-23T15:35:08Z","lastTransitionTime":"2023-06-23T15:28:59Z","message":"Cluster created successfully","reason":"ClusterCreated","status":"True","type":"Ready"}],"message":"Cluster created successfully","observedGeneration":2,"phase":"Ready","state":"Created","storage":{"deviceClasses":[{"name":"hdd"}]},"version":{"image":"quay.io/ceph/ceph:v17","version":"17.2.6-0"}}
```

let me know if you could think of something

**Comment by travisn** [comment]
I have a thought, but let's actually close this issue since it is resolved already for the `restore-quorum` command. The new issue is only for `rook status`.
